MASH-FRET/restruct_trace_file.m

```diff
@@ -1,36 +1,82 @@
-function restruct_trace_file(pname,varargin)
+function restruct_trace_file(varargin)
 % | Format trajectories files (*.txt) to MASH-importable structure.
 % |
 % | command: restruct_trace_file(pname);
-% | pname >> source directory
-% | varargin >> optional: DONOR_EXC 1-by-(nFRET+nS) laser wavelengths for 
+% | 1st argument (optional) >> source directory
+% | 2nd argument (optional) >> 1-by-(nFRET+nS) laser wavelengths for 
 % |             donor- and emitter-specific illuminations used in FRET and 
 % |             stoichiometry calculations.
 % |             example: if files contain columns for FRET_Cy3>Cy5, 
 % |             discrFRET_Cy3>Cy5, FRET_Cy5>Cy7, discrFRET_Cy5>Cy7,
 % |             S_Cy3, discrS_Cy3, S_Cy5, and discrS_Cy5
-% |             in order of appearence, DONOR_EXC = [532 532 638 638 532 
+% |             in order of appearence, RATIO_EXC = [532 532 638 638 532 
 % |             532 638 638]
 % |
 % | example: restruct_trace_file('C:\MyDataFolder\experiment_01\traces_processing\traces_ASCII\');
 
-% Last update: 3rd of April 2019 by M�lodie Hadzic
+% Last update: 10th of April 2019 by M�lodie Hadzic
+% --> implement GUI-based user input to select source directory, choose 
+%     file format (Trace processing/Transition analysis import) and
+%     ratio-specific illumination wavelength
+%
+% update: 3rd of April 2019 by M�lodie Hadzic
 % --> implement restructuration of files with multiple FRET and 
 %     stoichiometry and write substitute to MATLAB built-in function
 %     extractBetween for version older than 2016b
 %
 % update: 18th of February 2019 by M�lodie Hadzic
 % --> update help section
 
-% correct source directory
+% added by MH, 10.4.2019
+% get source directory
+if ~isempty(varargin) && numel(varargin)>=1
+    pname = varargin{1};
+else
+    pname = uigetdir();
+    if isempty(pname) || ~sum(pname)
+        return;
+    end
+end
+
 if ~strcmp(pname(end),'\')
     pname = cat(2,pname,filesep);
 end
 
-if ~isempty(varargin)
-    DONOR_EXC = varargin{1};
+% added by MH, 10.4.2019
+cd(pname);
+
+% modified by MH, 10.4.2019
+% get ratio-specific illuminations
+% if ~isempty(varargin)
+%     DONOR_EXC = varargin{1};
+% else
+%     DONOR_EXC = 532;
+% end
+if ~isempty(varargin) && numel(varargin)>=2
+    RATIO_EXC = varargin{2};
+else
+    RATIO_EXC = [];
+end
+
+% added by MH, 10.4.2019
+% activate/deactivate reorganization of ratio data
+if isempty(RATIO_EXC)
+    % ask user the new format of restructured files
+    module = questdlg({cat(2,'How do you want to format restructured trace ',...
```

Comment:
So in the first case (i.e. for trace processing) the (discretized) FRET columns are omitted, right?

Reply:
Yes, or more precisely, only intensities are imported (used in TP)